External/FEXCore/include/FEXCore/Utils/ThreadPoolAllocator.h

```diff
@@ -0,0 +1,160 @@
+#pragma once
+
+#include <atomic>
+#include <FEXCore/Utils/Allocator.h>
+
+#include <chrono>
+#include <cstddef>
+#include <list>
+#include <mutex>
+
+namespace FEXCore::Utils {
+  class ThreadPoolAllocator final {
+    public:
+      template<typename T>
+      struct AllocationInfo {
+        T Ptr;
+        size_t Size;
+      };
+
+      using ClockType = std::chrono::high_resolution_clock;
+
+      struct MemoryBuffer {
+        void* Ptr;
+        size_t Size;
+        std::chrono::time_point<ClockType> LastUsed;
+        std::atomic<bool> ClientClaimed;
+        std::atomic<bool> *CurrentClientOwnedFlag{};
+      };
+
+      using ContainerType = std::list<MemoryBuffer*>;
```

Comment:
Would `std::vector` work here? Looks like we're mostly just iterating over these (where `vector` provides better cache locality) and insertion/erasure is exceptional.

Reply:
I very explicitly chose list here so iterators aren't invalidated on insert/delete.
Could add a comment stating why I chose list.

Reply:
Added a comment that I'm requiring iterators to stay valid while manipulating the container. Resolved.